examples/react-native/src/lib/api.ts

```diff
@@ -0,0 +1,39 @@
+import movies from '../data/movies.json';
+
+export type Movie = {
+  title: string;
+  year: number;
+};
+
+export type MovieDetails = Movie & {
+  info: {
+    plot: string;
+    actors: string[];
+  };
+};
+
+function delay(t: number, v: () => void) {
+  return new Promise(function (resolve) {
+    setTimeout(resolve.bind(null, v), t);
+  });
+}
+
+export function fetchMovies(): Promise<Movie[]> {
+  console.log('fetchMovies');
+  return delay(200 + Math.floor(Math.random() * 2000), () =>
+    movies
+      .slice(0, 100)
+      .map((movie) => ({ title: movie.title, year: movie.year }))
+  ) as Promise<Movie[]>;
+}
+
+export function fetchMovie(title: string): Promise<MovieDetails> {
```

Comment:
lets make these `async` functions, which should get rid of the type assertion to a Promise because async functions always return a Promise.

Reply:
I meant removing `as Promise<Movie[]>;` - or why do we need that?

Reply:
We need that because movies.json is not the same type as Movie[]. TS seems not capable to infer this.

Reply:
oh okay, I didn't see that